tools/istio-iptables/pkg/dependencies/implementation.go

```diff
@@ -15,12 +15,46 @@
 package dependencies
 
 import (
+	"bytes"
 	"fmt"
 	"os"
 	"os/exec"
 	"strings"
+
+	"istio.io/istio/tools/istio-iptables/pkg/constants"
+)
+
+// XTablesExittype is the exit type of xtables commands.
+type XTablesExittype int
+
+// Learn from `xtables_exittype` of iptables.
+// `XTF_ONLY_ONCE`, `XTF_NO_INVERT`, `XTF_BAD_VALUE`, `XTF_ONE_ACTION` will eventually turned out to be a
+// parameter problem with explicit error message. Thus, we do not need to support them here.
+const (
+	// XTablesOtherProblem indicates a problem of other type in xtables
+	XTablesOtherProblem XTablesExittype = iota + 1
+	// XTablesParameterProblem indicates a parameter problem in xtables
+	XTablesParameterProblem
+	// XTablesVersionProblem indicates a version problem in xtables
+	XTablesVersionProblem
+	// XTablesResourceProblem indicates a resource problem in xtables
+	XTablesResourceProblem
 )
 
+var exittypeToString = map[XTablesExittype]string{
+	XTablesOtherProblem:     "xtables other problem",
+	XTablesParameterProblem: "xtables parameter problem",
+	XTablesVersionProblem:   "xtables version problem",
+	XTablesResourceProblem:  "xtables resource problem",
+}
+
+// XTablesCmds lists all the xtables related commands currently supported.
+var XTablesCmds = []string{
+	constants.IPTABLES, constants.IP6TABLES,
+	constants.IPTABLESRESTORE, constants.IP6TABLESRESTORE,
+	constants.IPTABLESSAVE, constants.IP6TABLESSAVE,
+}
+
```

Comment:
You only search for existence in this collection, so a set would be more efficient:

```suggestion
// XTablesCmds is the set of all the xtables-related commands currently supported.
var XTablesCmds map[string]bool{
	constants.IPTABLES: true,
	constants.IP6TABLES: true,
	constants.IPTABLESRESTORE: true,
	constants.IP6TABLESRESTORE: true,
	constants.IPTABLESSAVE: true,
	constants.IP6TABLESSAVE: true,
}

```

Reply:
Great advice, used a set instead. Thanks!